docs/reference/remap.cue

```diff
@@ -52,17 +51,589 @@ remap: {
+				"Pogs"]`.
+
+				### Combined
+
+				All of the path methods above can be combined in any way. Here's an example of a
+				complex path:
+
+				```
+				.transaction.(metadata | info).orders[0] = "a1b2c3d4e5f6"
+				```
+
+				This sets the first element of `.transaction.metadata.orders` to `"a1b2c3d4e5f6"`
+				*if* `.transaction.metadata` exist; if not, it sets `.transaction.info.orders` to
+				that value.
+				"""
+			examples: [
+				".",
+				".status_code",
+				#".message.event."time of occurrence""#,
+				".transaction.id",
+				".user.hobbies[0].description",
+				".event.(time | timestamp).format",
+				#""#,
+			]
+		}
+
+		"Expressions": {
+			href: "expressions"
+
+			description: """
+				*All* expressions in TRL resolve to a value. Expressions come in four kinds, listed
+				below, each of which resolves in a different way:
+
+				Expression type | Resolves to
+				:---------------|:-----------
+				Assignment | The assigned value
+				Control flow statements | The value returned by the chosen expression
+				Boolean expressions | The returned Boolean value
+				Blocks | The value returned by the last expression in the block
+				"""
+
+			in_n_out: [
+				{in: #".request_id = "a1b2c3d4e5f6""#, out:                      "string"},
+				{in: #"if (starts_with("v1", .version)) { .version = 1 }"#, out: "integer"},
+				{in: #"contains("emergency", "this is an emergency")"#, out:     "boolean"},
+				{
+					in: """
+						$is_success = { $code = .status_code; del(.status_code); $code == 200 }
+						"""
+					out: "boolean"
+				},
+			]
+		}
+
+		"Lines": {
+			href: "lines"
+
+			description: #"""
+				TRL expressions can be split across multiple lines using a backslash (`\`):
+
+				```
+				del(.one, .two, .three, .four \
+					.five, .six)
+				```
+
+				This statement is semantically identical to `del(.one, .two, .three, .four, .five,
+				.six)`.
+
+				Conversely, multiple expressions can be collapsed into a single line using a
+				semicolon (`;`) as the separator:
+
+				```
+				$success_code = 200; .success = .success_code == $success_code; del(.success_code)
+				```
+
+				You can also use line collapsing via semicolon in [control flow
+				statements](#control-flow).
+				"""#
+		}
+
+		"Functions": {
+			href: "functions"
+
+			description: """
+				In TRL, functions can take inputs (or no input) and return either a value or, in the
+				case of some functions, an error.
+				"""
+
+			examples: [
+				"parse_json(.message)",
+				"assert(.status_code == 500)",
+				#"ip_subnet(.address, "255.255.255.0")"#,
+				".request_id = uuid_v4()",
+			]
+		}
+
+		"Control flow": {
+			href: "control-flow"
+
+			description: """
+				TRL supports control flow operations using `if`, `else if`, and `else`. These can
+				be called on any expression that returns a Boolean. Here's a generic example of the
+				syntax:
+
+				```
+				if (condition) {
```

Comment:
The parenthesis around condition are only required in certain situations:

* If you provide a single "boolean expression", you don't need them (this includes equality checks, arithmetic, function calls, literal boolean values, path/variable queries).
* If you do anything else (such as an assignment, or when using multiple expressions separated by semicolons) you do need them